supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/AnalysisOfEndpointConnectionsClient.ts

```diff
@@ -0,0 +1,54 @@
+import * as ts from 'typescript';
+import * as fs from 'node:fs';
+
+// Get the file names from the command line arguments
+const fileNames = process.argv.slice(2);
+const HTTP_METHODS = ['get', 'post', 'put', 'delete'];
+
+for (const fileName of fileNames.filter(fileName => fileName.endsWith('.ts')))  {
+    // Load the TypeScript file
+    const sourceFile = ts.createSourceFile(fileName, fs.readFileSync(fileName).toString(), ts.ScriptTarget.ES2022, true);
+
+    // Store class property definitions
+    const classProperties: { [key: string]: string } = {};
+
+    // Start traversing the AST from the root
+    visit(sourceFile, classProperties, sourceFile, fileName);
+};
+
+// This function will be called for each node in the AST
+function visit(node: ts.Node, classProperties: { [key: string]: string }, sourceFile: ts.SourceFile, fileName: string) {
+    if (ts.isPropertyDeclaration(node) && node.initializer && ts.isStringLiteral(node.initializer)) {
+        const name = node.name.getText();
+        const value = node.initializer.getText().slice(1, -1); // Remove the quotes
+        classProperties[name] = value;
+    }
+
+    if (ts.isCallExpression(node)) {
+        const expression = node.expression;
+        // Check if the expression is a property access expression (e.g. httpClient.get)
+        if (ts.isPropertyAccessExpression(expression)) {
+            const name = expression.name.getText();
+            // Check if the property name is one of the httpClient methods
+            if (HTTP_METHODS.includes(name)) {
```

Comment:
You should check, whether the object, on which you call one of the `HTTP_METHODS` is a `httpClient` object.